core/capabilities/remote/target/caller.go

```diff
@@ -0,0 +1,135 @@
+package target
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	commoncap "github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/remote"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/remote/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+// remoteTargetCaller/Receiver are shims translating between capability API calls and network messages
+type remoteTargetCaller struct {
+	lggr                 logger.Logger
+	remoteCapabilityInfo commoncap.CapabilityInfo
+	localDONInfo         capabilities.DON
+	dispatcher           types.Dispatcher
+	requestTimeout       time.Duration
+
+	requestIDToExecuteRequest map[string]*callerExecuteRequest
+	mutex                     sync.Mutex
+}
+
+var _ commoncap.TargetCapability = &remoteTargetCaller{}
+var _ types.Receiver = &remoteTargetCaller{}
+
+func NewRemoteTargetCaller(ctx context.Context, lggr logger.Logger, remoteCapabilityInfo commoncap.CapabilityInfo, localDonInfo capabilities.DON, dispatcher types.Dispatcher,
+	requestTimeout time.Duration) *remoteTargetCaller {
+
+	caller := &remoteTargetCaller{
+		lggr:                      lggr,
+		remoteCapabilityInfo:      remoteCapabilityInfo,
+		localDONInfo:              localDonInfo,
+		dispatcher:                dispatcher,
+		requestTimeout:            requestTimeout,
+		requestIDToExecuteRequest: make(map[string]*callerExecuteRequest),
+	}
+
+	go func() {
```

Comment:
Can we convert to services.Service style with Start()/Stop() for consistency with other objects that launch their own coroutines?